### PR TITLE
Add IdentityMultiple

### LINKED ISSUE #44: Create systems using UniformScaling
See [this comment](https://github.com/JuliaReach/MathematicalSystems.jl/issues/40#issuecomment-446040193).

### PR DESCRIPTION
**Pull request by mforets**
Closes #44.

Tests to be added after merging with #43.

### CODE REVIEW

File: src/identity.jl [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+is needed for dispatch. The difference between `UniformScaling` and a
+`IdentityMultiple` is that while the size of the former is generic, the size of
+the latter is fixed.
+"""
+struct IdentityMultiple{T} <: AbstractMatrix{T}
+    M::UniformScaling{T}
+    n::Int
+end
+
+import Base:size, IndexStyle, getindex, setindex!, +, *, show
```
Comment (blegat) [inline src/identity.jl]: I am in favor of not importing it and writing e.g. `Base.:(*)` when adding a method

Comment (schillic) [reply src/identity.jl]: Is there a difference?

Comment (blegat) [reply src/identity.jl]: Just in term of readability I think. When looking at the method overload it is not clear of it is extending the Base function or creating a new one and I need to figure out if there was an `import`

Comment (schillic) [reply src/identity.jl]: Oh, so there is a difference :+1: (I thought that you would have to write `Base.:(*)` in both cases.)

Comment (mforets) [reply src/identity.jl]: Alright, I've used `Base.:..` style in `identity.jl`.

**Comment by mforets** [comment]
I've added the test from #40. The system is constructed properly, but the printing of an identity has problems; i would like to defer it to a new issue. For example:

```julia
julia> using MathematicalSystems, LazySets
julia> A, B = rand(4, 4), rand(4, 2);
julia> X = rand(Hyperrectangle, dim=4);
julia> U = rand(Ball2, dim=2);
julia> I4 = MathematicalSystems.IdentityMultiple(1.0*I, 4);
julia> s = ConstrainedLinearControlContinuousSystem(A, I4, X, B*U); # ok

julia> s  # error printing
julia> s
ConstrainedLinearControlContinuousSystem{Float64,Array{Float64,2},MathematicalSystems.IdentityMultiple{Float64},Hyperrectangle{Float64},LazySets.LinearMap{Float64,Bal
l2{Float64},Float64,Array{Float64,2}}}([0.823635 0.732318 0.959561 0.160496; 0.646987 0.587626 0.0463386 0.743196; 0.278478 0.535347 0.809034 0.481342; 0.172735 0.591
7 0.805676 0.385615], Error showing value of type ConstrainedLinearControlContinuousSystem{Float64,Array{Float64,2},MathematicalSystems.IdentityMultiple{Float64},Hype
rrectangle{Float64},LazySets.LinearMap{Float64,Ball2{Float64},Float64,Array{Float64,2}}}:
ERROR: MethodError: no method matching UnitRange{Int64}(::Int64)
Closest candidates are:
  UnitRange{Int64}(::Any, ::Any) where T<:Real at range.jl:255
  UnitRange{Int64}(::UnitRange{T<:Real}) where T<:Real at range.jl:846
  UnitRange{Int64}(::UnitRange) where T<:Real at range.jl:847
  ...
Stacktrace:
 [1] _show_nonempty(::IOContext{REPL.Terminals.TTYTerminal}, ::MathematicalSystems.IdentityMultiple{Float64}, ::String) at ./arrayshow.jl:365
 [2] show(::IOContext{REPL.Terminals.TTYTerminal}, ::MathematicalSystems.IdentityMultiple{Float64}) at ./arrayshow.jl:421

 [3] show_default(::IOContext{REPL.Terminals.TTYTerminal}, ::Any) at ./show.jl:332
 [4] show at ./show.jl:315 [inlined]
 [5] show(::IOContext{REPL.Terminals.TTYTerminal}, ::MIME{Symbol("text/plain")}, ::ConstrainedLinearControlContinuousSystem{Float64,Array{Float64,2},MathematicalSyste
ms.IdentityMultiple{Float64},Hyperrectangle{Float64},LazySets.LinearMap{Float64,Ball2{Float64},Float64,Array{Float64,2}}}) at ./sysimg.jl:194
 [6] display(::REPL.REPLDisplay, ::MIME{Symbol("text/plain")}, ::Any) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:13
1
 [7] display(::REPL.REPLDisplay, ::Any) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:135
 [8] display(::Any) at ./multimedia.jl:287
 [9] #invokelatest#1 at ./essentials.jl:697 [inlined]
 [10] invokelatest at ./essentials.jl:696 [inlined]
 [11] print_response(::IO, ::Any, ::Any, ::Bool, ::Bool, ::Any) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:154
 [12] print_response(::REPL.AbstractREPL, ::Any, ::Any, ::Bool, ::Bool) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:
139
 [13] (::getfield(REPL, Symbol("#do_respond#40")){Bool,getfield(REPL, Symbol("##50#59")){REPL.LineEditREPL,REPL.REPLHistoryProvider},REPL.LineEditREPL,REPL.LineEdit.P
rompt})(::Any, ::Any, ::Any) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:713
 [14] #invokelatest#1 at ./essentials.jl:697 [inlined]
 [15] invokelatest at ./essentials.jl:696 [inlined]
 [16] run_interface(::REPL.Terminals.TextTerminal, ::REPL.LineEdit.ModalInterface, ::REPL.LineEdit.MIState) at /Users/osx/buildbot/slave/package_osx64/build/usr/share
/julia/stdlib/v1.0/REPL/src/LineEdit.jl:2268
 [17] run_frontend(::REPL.LineEditREPL, ::REPL.REPLBackendRef) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:1034
 [18] run_repl(::REPL.AbstractREPL, ::Any) at /Users/osx/buildbot/slave/package_osx64/build/usr/share/julia/stdlib/v1.0/REPL/src/REPL.jl:191
 [19] (::getfield(Base, Symbol("##719#721")){Bool,Bool,Bool,Bool})(::Module) at ./logging.jl:309
 [20] #invokelatest#1 at ./essentials.jl:697 [inlined]
 [21] invokelatest at ./essentials.jl:696 [inlined]
 [22] macro expansion at ./logging.jl:306 [inlined]
 [23] run_main_repl(::Bool, ::Bool, ::Bool, ::Bool, ::Bool) at ./client.jl:334
 [24] exec_options(::Base.JLOptions) at ./client.jl:246
 [25] _start() at ./client.jl:425
```

**Comment by schillic** [comment]
What is the problem here?

**Comment by mforets** [comment]
> i do not understand why travis was failing.

it was because `UniformScaling` lives in `LinearAlgebra`, and i got `using LinearAlgebra` in my `startup.jl` :)

**Pull request merged**